src/mixins/Regions.js

```diff
@@ -86,12 +86,37 @@ const RegionsMixin = types
 
       const result = regions.filter(region => {
         if (excludeSelf && region === self) return false;
-        return region.dynamic && region.type === type && region.labelName === labelName;
+        const canBePartOfNotification = self.supportSuggestions ? self.dynamic : true;
+
+        return canBePartOfNotification
+          && region.type === type
+          && region.labelName === labelName
+          && region.results?.[0]?.to_name === self.results?.[0]?.to_name;
```

Comment:
It potentially compare `undefined === undefined` here. Could it lead to any unexpected issues?

Reply:
I've checked that. It seems that the only way to get `undefined === undefined` is collecting connected regions in the case with config contains two classification tags with the same `to_name`.  I'm not sure if wee need to group them but it's anyhow is better than it was.